### PR TITLE
feat: Ad page level configs

### PR DESCRIPTION
**Pull request by nehasri89**
Each ad sends out some targetting config to Ad server so that it can receive the most appropriate Ad on the page. This PR adds the page level config for the ads on web. 
https://nidigitalsolutions.jira.com/wiki/spaces/PD/pages/154796078/Advertising+web+spec

The population of the config is dealt in render side in another PR https://github.com/newsuk/cps-content-render/pull/2176

### CODE REVIEW

File: packages/ad/ad.stories.js [hunk 1]
```diff
@@ -39,31 +39,45 @@ const withOpenInNewWindow = children => {
...
     withOpenInNewWindow(
       <View>
-        <Ad section="article" code="ad-header" pos="header" />
-        <Ad section="article" code="intervention" pos="inline-ad" />
-        <Ad
-          section="article"
-          code="ad-article-sponsored"
-          pos="article-sponsored-ad"
-        />
+        <AdComposer>
```
Comment (craigbilner) [inline packages/ad/ad.stories.js]: why wouldn't there be just one `AdComposer`?

File: packages/ad/__tests__/ad.test.js [hunk 2]
```diff
@@ -60,7 +82,15 @@ describe("Ad", () => {
...
   it("hides the placeholder when the ad is ready", () => {
-    const component = shallow(<Ad {...adProps} code="ad-header" />);
+    jest.spyOn(console, "error").mockImplementation(() => {});
+    const mockAdConfig = {
+      networkId: "25436805",
+      adUnit: "d.thetimes.co.uk",
+      pageTargeting: {}
+    };
+    const component = shallow(
+      <Ad {...adProps} overrideAdConfig={mockAdConfig} code="ad-header" />
```
Comment (craigbilner) [inline packages/ad/__tests__/ad.test.js]: should it be called `override`, don't think the consumer cares, it should just be a config that happens to override?

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/b37d2b56dcef0d5fc878d8d9384cceac6f39959c/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/e1e2394b6f157dae9c921ffef2486d66ac4e42fb/dextrose-test/index.html

**Pull request merged**